tensorflow/dtensor/mlir/utils/collective_lowering.cc

```diff
@@ -368,6 +381,150 @@ mlir::Operation* EmitCollectiveReduceScatter(
   return collective_reduce_scatter;
 }
 
+mlir::Operation* EmitCollectiveAllToAll(
+    mlir::OpBuilder& builder, const mlir::Location& loc, mlir::Value input,
+    const mlir::DenseIntElementsAttr& group_assignment, int32 concat_dimension,
+    int32 split_dimension, int32 key_base, mlir::Value device_id,
```

Comment:
Thanks a lot for making this signature aligned with XlaAllToAll!

Reply:
Thanks! I was also debating whether to make `DTensorAllToAll`'s signature also match with XLA, but it was easier and more algined with the existing DTensor ops to use target/source layout instead.